Python/Product/PythonTools/PythonTools/Extensions.cs

```diff
@@ -911,5 +918,32 @@ public static class Extensions {
             }
             return prettyPrinted.ToString().Trim();
         }
+
+        public static void RunSynchronouslyOnUIThread(this System.Threading.Tasks.Task task, IServiceProvider serviceProvider, [CallerMemberNameAttribute] string callerMemberName = "") {
+            if (!Dispatcher.CurrentDispatcher.CheckAccess()) {
+                string message = string.Format("{0} must be called on a UI thread.", callerMemberName);
+                Debug.Fail(message);
+                return;
```

Comment:
So in that case, we really want this in VsTaskExtensions as an alternative to `WaitAndUnwrapExceptions` that just waits directly when off the UI thread and does this when on the UI thread. Then anywhere we can pass in a service provider we should, so that we get the right behavior.